### ISSUE TITLE
Clarification on Focus Not Obscured's "user-opened content"

### ISSUE THREAD

**Issue by fstrr** [w3c/wcag#3333]
I'm having trouble understanding "without actually altering the keyboard focus" in the User-Opened Content section of [Focus Not Obscured (Minimum)](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum.html):

> In recognition of more complex applications, the note allows for user-opened content to obscure the item receiving focus, provided the user can bring the item with focus into view without actually altering the keyboard focus. Keyboard actions that may allow the item with focus to be revealed include: 1) using Esc to dismiss the opened content, 2) using keys to scroll the content in the viewport to reveal the item with focus, or 3) issuing a key to move between overlays.

Let's say:

1. I have a chatbot that is closed/collapsed on page load, but I've opened it so it's expanded.
2. The expanded chatbot's interface completely obscures some interactive content on the underlying page.
3. I decide to keyboard-navigate out of the chatbot and tab onto a link that's completely obscured by the chatbot's interface.

Is "without actually altering the keyboard focus" saying that, when I'm focused on an item I've caused to be completely obscured, I need to be able to either collapse the chatbot when I'm not in it or use a non-tab-key method to scroll the obscured content into view? If it is saying that, would this be clearer?:

------

In recognition of more complex applications, the note allows for user-opened content to obscure the item receiving focus, provided the user can bring the item with focus into view using a method that doesn't require navigating back to the user-opened content to close it. For example: keyboard actions that may allow the item with focus to be revealed include:

1. using the Escape key to dismiss the opened content;
2. using keys to scroll the content in the viewport to reveal the item with focus;
3. issuing a key to move between overlays.

For example:

* A user expands a chatbot's interface, which results in content on the underlying page being fully obscured. The user uses their keyboard to navigate to a link that has been fully obscured by the open chatbot. They press the Escape key to close the chatbot, which un-obscures the link.
* A user expands a fixed-position page feedback component at the bottom of a Web page. They then use their keyboard to navigate to a link that's fully obscured by the expanded component and press the down arrow on their keyboard to scroll the content on the page until the link is un-obscured.
* A user opens a web-based multiuser authoring application. An overlay appears displaying a list of people who have contributed to the document. The user tabs through the list of contributors and activates one of them. The application displays a new overlay, which obscures the first one, that displays that person's recent contributions. The user presses the F6 key to toggle the stacking order of the two overlays.

**Comment by scottaohara** [comment]
Some additional wording adjustments covering the proposed above changes, as well as some other changes for the section content preceding that modified above:

>This SC contains a note regarding content that is opened or disclosed by the user. One example of such content is a menu button opened by a user that presents a list of choices visually displayed over pre-existing content on the screen. Such content can obscure other information on the screen, but it does not obscure an item receiving keyboard focus, because the list of choices auto-dismisses after keyboard focus has been moved to another focusable element. However, authors may create user-opened content that is intentionally designed to persist until closed by the user, such as a chat window or other popover content which may need to be viewed alongside other content of the page. Such persistent content has the potential to fail Focus Not Obscured (Minimum). Various types are described in this section. All can be designed so that they pass this Success Criterion.
>
> In recognition of more complex interfaces and user needs, the note allows for user-opened content to obscure the item receiving focus, provided the user can bring the item with focus into view using a method that doesn't require navigating back to the user-opened content to dismiss it. For example: keyboard actions that may allow the item with focus to be revealed include:
>
> 1. using the Escape key to dismiss the obscuring content;
> 1. using keys to scroll the content in the viewport to reveal the item with focus;
> 1. issuing a key to move between overlays.
>
>For example:
>
>- A user invokes a chat interface as a popover dialog, which results in some content of the underlying page being fully obscured. The user navigates away from the chat interface by use of the tab key, focusing a link that has been fully obscured by the dialog. The user presses the Escape key to dismiss the chat interface, which un-obscures the link.
>- A user expands a fixed-position page feedback component at the bottom of a Web page. They then use their keyboard to navigate to a link that's fully obscured by the expanded component and press the down arrow or space key on their keyboard to scroll the content on the page, un-obscuring the link.
>- A user opens a web-based multiuser authoring application. An overlay appears displaying a list of people who have contributed to the document. The user tabs through the list of contributors and activates one of them. The application displays a new overlay, which obscures the first one, that displays that person's recent contributions. The user presses the F6 key to toggle the stacking order of the two overlays.


I'm personally not opposed to these additional examples, but I do wonder if they belong in this section, or if they'd better fit into the User openable, persistent disclosures section - since that's what these examples represent.

Regarding that section in particular, I also think the following bullet needs to be modified

>When focus leaves the additional content, the additional content is hidden or collapsed

so that it references what is being mentioned in the modified suggested text:

>When focus leaves the additional content, the additional content is automatically hidden or collapsed, or the content can be hidden or collapsed by use of a dedicated keyboard command. E.g., the Escape key.

**Comment by fstrr** [comment]
I think that moving that content to the User openable, persistent disclosures section makes sense, as does the modification you mention. It sounds like the best thing to do might be to create a new branch and start work there. I'll give this thread a couple of days to see if there's any major feedback, and if there isn't, I'll start that branch.

**Comment by mbgower** [comment]
> Is "without actually altering the keyboard focus" saying that, when I'm focused on an item I've caused to be completely obscured, I need to be able to either collapse the chatbot when I'm not in it or use a non-tab-key method to scroll the obscured content into view?

Yes

> using a method that doesn't require navigating back to the user-opened content to close it

My concern with this phrase is that navigating back to the chat to dismiss it is totally valid if I, say, issue a dedicated key that moves between regions of the page. Doing so would of course also _move_ focus there, so I agree the current wording doesn't work, but this also doesn't work.

**Comment by scottaohara** [comment]
Agreed @mbgower 

The note in the spec text that says:
“Content opened by the user may obscure the component receiving focus. If the user can reveal the focused component without advancing the keyboard focus…”

Needs to be changed if it’s acceptable for someone to move keyboard focus back to the obscuring content to close it.

**Comment by fstrr** [comment]
> Yes, OR move to it [the currently obscured content] in some way that is not via the Tab key (which is the part that has poor phrasing).

By "move to it" do you mean "bring into focus"? Or something else? If I have a fixed-position chatbot that's obscuring some links in a sidebar, if I didn't want to automatically close the chatbot when it wasn't being used, I would have to change the z-index of the sidebar so it was above the chatbot, or something equally unlikely, to un-obscure the links.

**Comment by alastc** [comment]
DIscussion from the meeting: https://www.w3.org/2023/10/31-ag-minutes#x233
Question to Gundula: https://lists.w3.org/Archives/Public/public-wcag2-issues/2023Nov/0000.html

**Comment by mbgower** [comment]
Advancing focus needs to allow toggling between panes with their own tabring. The focus is switched between the panes. the focus is not advanced.

**Comment by fstrr** [comment]
[Interesting ARIA issue about not using the Escape key to dismiss custom tooltips](https://github.com/w3c/aria-practices/issues/1506). Could also be relevant to this.
